include/operations/CompoundOperation.hpp

```diff
@@ -158,5 +158,20 @@ namespace qc {
         [[nodiscard]] const auto& at(std::size_t i) const { return ops.at(i); }
 
         std::vector<std::unique_ptr<Operation>>& getOps() { return ops; }
+
+        [[nodiscard]] std::vector<dd::Qubit> getUsedQubits() const override {
+            std::vector<dd::Qubit> usedQubits{};
+            for (const auto& op: ops) {
+                for (auto target: op->getTargets()) {
+                    usedQubits.push_back(target);
+                }
+                for (auto control: op->getControls()) {
+                    usedQubits.push_back(control.qubit);
+                }
```

Comment:
this is not guaranteed to work properly. you have to call `getUsedQubits` on each sub-operation and then add build the union of the current set and the newly generated set. To this end, it is probably easiest and cleanest to just replace the `std::vector<dd::Qubit>` with a `std::set<dd::Qubit>` which handles sorting an uniqueness automatically.
If you really need a `vector` as the result of the call (which I don't think you need), then you can still construct a vector from the set by (pseudocode)
```c++
auto vec = std::vector{set.begin(), set.end()};
```

(the union of two `set`s can be easily be computed in-place by using `set1.insert(set2.begin(), set2.end())`)

Reply:
Indeed! I changed the code as suggested.